code/modules/mod/modules/modules_engineering.dm

```diff
@@ -151,3 +151,101 @@
 	QDEL_NULL(chain)
 	return ..()
 
+///Atmos water tank module
+
+/obj/item/mod/module/firefighting_tank
+	name = "MOD firefighting tank"
+	desc = "A refridgerated and pressurized module tank with an extinguisher nozzle, intended to fight fires. Swaps between extinguisher, nanofrost launcher, and metal foam dispenser for breaches. Nanofrost converts plasma in the air to nitrogen, but only if it is combusting at the time."
+	icon_state = "firefighting_tank"
+	module_type = MODULE_ACTIVE
+	complexity = 2
+	active_power_cost = DEFAULT_CHARGE_DRAIN * 3
+	device = /obj/item/extinguisher/mini/mod
+
+#define EXTINGUISHER 0
+#define NANOFROST 1
+#define METAL_FOAM 2
+
+/obj/item/extinguisher/mini/mod
+	name = "modsuit extinguisher nozzle"
+	desc = "A heavy duty nozzle attached to a modsuit's internal tank."
+	icon = 'icons/obj/watertank.dmi'
+	icon_state = "atmos_nozzle"
+	item_state = "nozzleatmos"
+	safety = 0
+	max_water = 500
+	power = 8
+	precision = 1
+	cooling_power = 5
+	w_class = WEIGHT_CLASS_HUGE
+	flags = NODROP //Necessary to ensure that the nozzle and tank never seperate
+	var/nozzle_mode = 0
+	var/metal_synthesis_cooldown = 0
+	var/nanofrost_cooldown = 0
+
+/obj/item/extinguisher/mini/mod/attack_self(mob/user as mob)
+	switch(nozzle_mode)
+		if(EXTINGUISHER)
+			nozzle_mode = NANOFROST
+			icon_state = "atmos_nozzle_2"
+			to_chat(user, "Swapped to nanofrost launcher")
+			return
+		if(NANOFROST)
+			nozzle_mode = METAL_FOAM
+			icon_state = "atmos_nozzle_3"
```

Comment:
Please remove the icon_state setting from here, and set them in the proc called update_icon_state(). Then at the end of this proc, call update_icon(UPDATE_ICON_STATE)
```suggestion
```